design-system/src/theme-provider.tsx

```diff
@@ -112,44 +109,18 @@ type TUseThemeResult = {
   ) => Old | New;
   isNewTheme: boolean;
 };
-const useTheme = (parentSelector = defaultParentSelector): TUseThemeResult => {
-  const [theme, setTheme] = useState<ThemeName>('default');
-  const parentSelectorRef = useRef(parentSelector);
-
-  const mutationChangeCallback = useCallback((mutationList) => {
-    // We expect only a single element in the mutation list as we configured the
-    // observer to only listen to `data-theme` changes.
-    const [mutationEvent] = mutationList;
-    setTheme((mutationEvent.target as HTMLElement).dataset.theme as ThemeName);
-  }, []);
-
-  useMutationObserver(parentSelector(), mutationChangeCallback, {
-    attributes: true,
-    attributeFilter: ['data-theme'],
-  });
-
+const useTheme = (_parentSelector = defaultParentSelector): TUseThemeResult => {
   const themedValue: TUseThemeResult['themedValue'] = useCallback(
-    (defaultThemeValue, newThemeValue) =>
-      theme === 'default' ? defaultThemeValue : newThemeValue,
-    [theme]
+    (_defaultThemeValue, newThemeValue) => newThemeValue,
+    []
   );
 
-  // If we use 'useLayoutEffect' here, we would be trying to read the
-  // data attribute before it gets set from the effect in the ThemeProvider
-  useEffect(() => {
-    // We need to read the current theme after the provider is rendered
-    // to have the actual selected theme (calculated client-side) in the
-    // hook local state
-    const nextTheme = parentSelectorRef.current()?.dataset.theme as ThemeName;
-    if (nextTheme) {
-      setTheme(nextTheme);
-    }
-  }, []);
-
   return {
-    theme,
+    theme: 'default',
+    // @deprecated
     themedValue,
-    isNewTheme: theme === 'test',
+    // @deprecated
+    isNewTheme: true,
```

Comment:
We will be returning a static theme value since we now have just one and the other two properties will be deprecated.
Another option is to directly removing: since we were only using them internally, I think it would be save, but I want to know your thoughts about it. It would mean that teams relying on them will be forced to update their code to be able to get any new `ui-kit` versions. That's why I think deprecating them first would be helpful.

Reply:
Updated in this commit: a20220d2b0974e85a814750710b1608f9b57f855

Thanks!